visits-service-new/src/test/java/com/petclinic/visits/visitsservicenew/DataLayer/VisitRepoTest.java

```diff
@@ -0,0 +1,146 @@
+package com.petclinic.visits.visitsservicenew.DataLayer;
+
+import org.reactivestreams.Publisher;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DataMongoTest
+public class VisitRepoTest {
+
+    @Autowired
+    private VisitRepo visitRepo;
+    Visit visit = buildVisit();
+
+
+
+    @Test
+    void findByVisitId(){
+        Publisher<Visit> visitPublisher = visitRepo.deleteAll().thenMany(visitRepo.save(visit));
+
+        StepVerifier.create(visitPublisher).expectNextCount(1).verifyComplete();
+
+        Mono<Visit> getVisit = visitRepo.findByVisitId(visit.getVisitId());
+
+        Publisher<Visit> comp = Mono.from(visitPublisher).then(getVisit);
+
+        StepVerifier.create(comp)
+                .consumeNextWith(gotVisit -> {
+
+                    assertEquals(visit.getVisitId(), gotVisit.getVisitId());
+                    assertEquals(visit.getPetId(), gotVisit.getPetId());
+                    assertEquals(visit.getDay(), gotVisit.getDay());
+                    assertEquals(visit.getMonth(), gotVisit.getMonth());
+                    assertEquals(visit.getYear(), gotVisit.getYear());
+                    assertEquals(visit.getPractitionerId(), gotVisit.getPractitionerId());
+                    assertEquals(visit.isStatus(), gotVisit.isStatus());
+                }).verifyComplete();
+    }
+    @Test
+    void findByPetId(){
```

Comment:
Have you thought about renaming this test to 'findVisitsByPetId'? Similar to the other tests names